### PR TITLE
Kusto language

### PR DESCRIPTION
**Pull request by kopytjuk**
Add kusto language support to `extra/`.

Fix path handling in `tools/lib/language.js` on windows machines.

### CODE REVIEW

**Comment by yyyc514** [comment]
We don't merge languages into core anymore because of lack of developer time to maintain them in the long-run.  If you make your own repository though you can make a PR that adds a link to it so those who'd like to use your language as a 3rd party plugin can easily do see.

**Comment by yyyc514** [comment]
> Fix path handling in tools/lib/language.js on windows machines.

If there is something here though we'd probably welcome a PR for a fix just for that.

File: tools/lib/language.js [hunk 1]
```diff
@@ -63,8 +63,7 @@ class Language {
   }
 
   static fromFile(filename) {
-    if (filename.startsWith("/") || filename.startsWith("."))
-    {
+    if (filename.startsWith("/") || filename.startsWith(".") || process.platform === "win32") {
```
Comment (yyyc514) [inline tools/lib/language.js]: This doesn't look correct though, what exactly is the problem you were trying to solve here?

Comment (kopytjuk) [reply tools/lib/language.js]: Windows file paths start with `C:\Users\...`  - without this platform check the file name becomes: 

```
/src/languages/C:\Users\...
```

Comment (yyyc514) [reply tools/lib/language.js]: Ah, that's not a correct fix though, that will cause it to break elsewhere.

You'd need to do something like this:

https://stackoverflow.com/questions/21698906/how-to-check-if-a-path-is-absolute-or-relative

**Pull request closed without merging**